history.py

```diff
@@ -0,0 +1,32 @@
+from catalog import models as cmod
+
+class LastFiveMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+        # One-time configuration and initialization.
+
+    def __call__(self, request):
+        # Code to be executed for each request before
+        # the view (and later middleware) are called.
+        request.session.get('last_five', [])
```

Comment:
you can also use '' like this --> request.session.get('last_five', '')

Reply:
Great idea. I did the "[]" so that it will initialize a list if the list wasn't previously there.